crates/dht/src/time.rs

```diff
@@ -200,6 +211,98 @@ impl PartitionedTime {
 
         Ok(())
     }
+
+    /// Inform the [PartitionedTime] that some ops have been stored.
+    ///
+    /// It is expected that the caller ensures that the ops belong to the hash range managed by this
+    /// [PartitionedTime]. This method will update the hashes of the full and partial slices that
+    /// the incoming ops belong to.
+    ///
+    /// If the op happens to be new enough to not belong in a slice, then it will be ignored. The
+    /// op will be discovered later when a partial slice update adds a slice that includes the op.
+    pub(crate) async fn inform_ops_stored(
+        &mut self,
+        store: DynOpStore,
+        stored_ops: Vec<StoredOp>,
+    ) -> K2Result<()> {
+        let full_slice_end = self.full_slice_end_timestamp();
+
+        for op in stored_ops {
+            if op.timestamp < full_slice_end {
+                // This is a historical update. We don't really expect this to happen too often.
+                // If we're syncing because we've been offline then it's okay and we should
+                // try to detect that when it's happening but otherwise it'd be good to log a warning
+                // here.
+
```

Comment:
So add a `eprintln!` for now?

Reply:
There is tracing, but there wasn't when I wrote myself a note, honest :) I've added a log message now